obsidian-tasks-group/obsidian-tasks
# feat: Optionally put next recurrence below current task

# Description



## Motivation and Context

The current implementation of recurrences pushes the completed task below and replaces the original task with the new recurrence (as documented).

In my workflow I prefer to have it the other way around  - the next recurrence of a recurring task on the line below the original task. This mimic "human" behaviour  - completing a task and making a new one on the line below.

So I made an option for that. And the default behaviour is the original one.

Please feel free to reformulate my English in the settings and/or docs as you can do it quite better =)

## How has this been tested?

Relevant user tests added.

Manual tests of the Settings Tab in my live vault:
1. With the default setting toggle an incomplete task with recurrence. Result : the new task appears on the original line and the completed one on the next line. Revert the changes in the file.
2. Set the setting to `true` (enabled) and toggle the same incomplete task with recurrence. Result : the new task appears below the original line and the original is at its place with DONE status. Revert the changes in the file.
3. Set the setting back to `false` (disabled, default) and toggle the same incomplete task with recurrence. Result : the new task appears on the original line and the completed one on the next line. Revert the changes in the file.

Task used for testing: `- [ ] #task blablabla 🔁 every week on Tuesday ⏳ 2023-05-23`

## Screenshots (if appropriate)

Below is the view with the enabled option which is not the default.

I decided not to make one whole section for this and put it in the `Date Settings`. Lmk if you prefer otherwise.

![Снимок экрана 2023-05-16 в 23 52 56](https://github.com/obsidian-tasks-group/obsidian-tasks/assets/93825870/3cc20c90-3104-40e8-9f46-386853215259)


## Types of changes



Changes visible to users:

- [ ] **Breaking change** (prefix: `feat!!` or `fix!!` - fix or feature that would cause existing functionality to not work as expected)
- [ ] **Sample vault** (prefix: `vault` - improvements to the [Tasks-Demo sample vault](https://github.com/obsidian-tasks-group/obsidian-tasks/tree/main/resources/sample_vaults/Tasks-Demo))
- [ ] **Contributing Guidelines** (prefix: `contrib` - any improvements to documentation content **for contributors** - see [Contributing to Tasks](https://publish.obsidian.md/tasks-contributing/))

Internal changes:

- [ ] **Infrastructure** (prefix: `chore` - examples include GitHub Actions, issue templates)

## Checklist





## Terms



- [x] My contribution follow this project's [contributing guide](https://publish.obsidian.md/tasks-contributing)
